### PR TITLE
Enable graceful shutdown of headless client

### PR DESCRIPTION
**Pull request by glitchfur**
Currently, the headless client cannot receive the terminate signal inside the container (when using `docker stop` for example) because it is not running as PID 1. This can be shown with `ps` (after installing `procps` in the container):

```
# docker exec neosvr-headless ps a
    PID TTY      STAT   TIME COMMAND
      1 pts/0    Ss+    0:00 /bin/sh -c /Scripts/setup_neosvr.sh && /Scripts/start_neosvr.sh
    102 pts/0    S+     0:00 /bin/sh /Scripts/start_neosvr.sh
    103 pts/0    Sl+    0:18 mono /home/steam/neosvr-headless/Neos.exe -c /Config/Config.json -l /Logs
```

PID 1 is taken by the script that launched the headless client. This pull request resolves this issue by utilizing `exec` statements in the scripts to make `mono` replace the shell process and inherit its PID:

```
# docker exec neosvr-headless ps a
    PID TTY      STAT   TIME COMMAND
      1 pts/0    Ssl+   0:11 mono /home/steam/neosvr-headless/Neos.exe -c /Config/Config.json -l /Logs
```

This PR also changes the stop signal from the default `SIGTERM` to `SIGINT`, because the headless client does not seem to respond to `SIGTERM` even as PID 1. `SIGINT` emulates a press of `Ctrl+C` at the console. In my testing, this made the headless client shut down but not exit its process, meaning Docker still ultimately kills the container with `SIGKILL`.  But at least this allows the headless client to perform a clean shutdown before that happens.

I've also made improvements to my previous PR (#1) by making the following changes:

- The script `setup_neosvr.sh` has been relegated to being an `ENTRYPOINT`. This means that child images based on this one no longer need to include `setup_neosvr.sh` in their `CMD` if they choose to override it.
- At the end of `setup_neosvr.sh`, any arguments passed (e.g. `CMD`) are executed verbatim.
  - By default, this is `start_neosvr.sh`.
- Any custom commands executed also get the benefit of being PID 1 so they will be terminated properly if the Docker container is stopped.

tl;dr: These changes allow `CMD` to be overridden without any "gotchas" and get the expected behavior, while maintaining the benefit that the headless client is downloaded at runtime.

### CODE REVIEW

**Comment by shadowpanther** [comment]
This is some advanced Docker witchery that I don't understand yet. I hope you've tested the changes locally to confirm that the image still functions as intended?

**Comment by glitchfur** [comment]
Yes. Your image without modification by child images still downloads and starts up the headless client as it did prior to this PR.

The modifications I'm suggesting are mainly to make it easier to base new Docker images on this one in the "expected" way, as I am doing myself for my own project. I believe that situation is isolated to just myself at the moment, considering I only submitted my initial PR yesterday, but at least for me personally I can confirm that child images work in the way that I'd expect them to as well.

What I mean by "expected" way here is that images should be able to override `CMD` completely in their image without hindering your image's ability to download the headless client at runtime, which is the case with this PR.
